plugins/scaffolder-backend/src/scaffolder/storage/disk.ts

```diff
@@ -59,43 +63,38 @@ export class DiskStorage implements Base {
   }
 
   private async index(): Promise<DiskIndexEntry[]> {
-    return new Promise((resolve, reject) => {
-      glob(`${this.repoDir}/**/template-info.json`, async (err, matches) => {
-        if (err) {
-          reject(err);
-        }
+    const matches = await globby(`${this.repoDir}/**/template-info.json`);
 
-        const fileContents: Array<{
-          location: string;
-          contents: string;
-        }> = await Promise.all(
-          matches.map(async (location: string) => ({
-            location,
-            contents: await fs.readFile(location, 'utf-8'),
-          })),
-        );
+    const fileContents: Array<{
+      location: string;
+      contents: string;
+    }> = await Promise.all(
```

Comment:
i think the number of templates is small enough to do everything in this method in a plain for loop, but it's fine to keep the optimisation if you like